### PR TITLE
fix: enable ssr when `build: true` is provided

### PR DESCRIPTION
**Pull request by antfu**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -23,9 +23,10 @@ function nuxtVite () {
     return
   }
 
-  // Disable SSR by default
+  // Disable SSR by on dev by default
   const ssrEnabled = nuxt.options.ssr && nuxt.options.vite?.ssr
-  if (!ssrEnabled) {
+  const isBuild = !nuxt.options.dev && viteOptions.build
```
Comment (danielroe) [inline src/index.ts:28]: @antfu What would you say the blockers are from using SSR in both dev & prod?

Comment (antfu) [reply src/index.ts:28]: Currently the build is done via `@vitejs/plugin-legacy`, which make the bundling speed and bundle size worse then Webpack. As discussed with @pi0 we might need to figure out an solution for this while been careful about this default change, so maybe we will enabled build by default in v0.3.x

Comment (pi0) [reply src/index.ts:28]: Sure. PS: We are using a [nasty workaround](https://github.com/nuxt/vite/blob/8d2209ddb0e2b92bcc33af4d49f26a3a4c9b6f90/src/server.ts#L26) for vue2 plugin's SSR support also preload/prefetch needs module id augmentation (same as css-loader fork for vue2 and vue-loader)

**Comment by pi0** [comment]
Note: This is a workaround since the current `build: true` flag for production mode, does not support SPA only build (`ssr: false). (#185)

I'm pending this PR to check if there is a quick solution to #185 and not hiding problem but if not, we shall proceed with this for a quick fix.

**Comment by pi0** [comment]
Let's have proper SSR support and also fix SPA issue in the meanwhile without workaround :)

**Pull request closed without merging**